src/graphql/root/mutation/captcha-request-auth-code.ts

```diff
@@ -0,0 +1,54 @@
+import { GT } from "@graphql/index"
+import verifyCaptchaAndReturnOTP from "@core/auth-code-request"
+
+import Phone from "@graphql/types/scalar/phone"
+import SuccessPayload from "@graphql/types/payload/success-payload"
+
+const CaptchaRequestAuthCodeInput = new GT.Input({
+  name: "CaptchaRequestAuthCodeInput",
+  fields: () => ({
+    phone: { type: GT.NonNull(Phone) },
+    challengeCode: { type: GT.NonNull(GT.String) },
+    validationCode: { type: GT.NonNull(GT.String) },
+    secCode: { type: GT.NonNull(GT.String) },
+  }),
+})
+
+const CaptchaRequestAuthCodeMutation = GT.Field({
+  type: GT.NonNull(SuccessPayload),
+  args: {
+    input: { type: GT.NonNull(CaptchaRequestAuthCodeInput) },
+  },
+  resolve: async (_, args, { logger, ip, geetest }) => {
+    const {
+      phone,
+      challengeCode: geetestChallenge,
+      validationCode: geetestValidate,
+      secCode: geetestSeccode,
+    } = args.input
+
+    try {
+      // TODO: redo this with use-case and errors pattern
+      await verifyCaptchaAndReturnOTP({
```

Comment:
not all fails throw an exception, if `geetest.validate` fail it returns false

Reply:
There is an if statement in `verifyCaptchaAndReturnOTP` for that case

Reply:
yes, that is the point, it returns false (not an exception) and in that case we are returning `success: true`